### ISSUE TITLE
Fight command doesn't deselect after issuing it with Right Click (for single unit)

### ISSUE THREAD

**Issue by sprEEEzy** [beyond-all-reason/Beyond-All-Reason#2314]
Problematic behavior of fight (and patrol) when using it with a single unit. The command cannot be deselected with right click (unless your mouse is completely still), and it just issues chains of fight commands like you're holding shift (but you're not). This behavior seems technically useless for fight & patrol.

https://github.com/beyond-all-reason/Beyond-All-Reason/assets/9414375/5f3247bf-6652-4878-b223-3d822ccd0aea

Solution: turn all right click features off for fight/patrol when only one unit selected, make it only deselect.

**Comment by sprEEEzy** [comment]
Unfortunately not. Same behavior with that widget off.

**Comment by hihoman23** [comment]
Also, the same problem exists for attack too(and probably more commands that I don't know of(resurrect?))

**Comment by sprEEEzy** [comment]
Correct. When I say Fight in my reports, I also mean all similar commands that suffer from same issues.

**Comment by sprEEEzy** [comment]
I'd like this issue to be reopened, as it isn't fully fixed.

If you're moving your cursor while clicking right click, it will still issue the command, which isn't the desired behavior.

**Comment by icexuick** [comment]
Seems to only be the case when you hold shift.

**Comment by sprEEEzy** [comment]
Yes.

By the way, I also cannot get "Repeat command when dragging..." to save to OFF between games. It always defaults on ON.

**Comment by sprEEEzy** [comment]
Another issue. When having multiple units selected, fight right click has inconsistent behavior.

1. If you keep the cursor in place and fight RMB, it will deselect.
2. If you're moving the cursor and fight RMB, it will issue the formation fight, and then always deselect it.
